ipc-queue/src/position.rs

```diff
@@ -0,0 +1,71 @@
+/* Copyright (c) Fortanix, Inc.
+ *
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+use super::*;
+use std::sync::atomic::Ordering;
+
+/// `PositionMonitor<T>` can be used to record the current read/write positions
+/// of a queue. Even though a queue is comprised of a limited number of slots
+/// arranged as a ring buffer, we can assign a position to each value written/
+/// read to/from the queue. This is useful in case we want to know whether or
+/// not a particular value written to the queue has been read.
+pub struct PositionMonitor<T: 'static> {
+    pub(crate) read_epoch: Arc<AtomicU64>,
```

Comment:
What's the benefit of avoiding it? From what I understand, we need this function in the crate in one form or another. We could introduce a function inside the `impl<T> PositionMonitor<T> {}` block that does the same, but what would be the win in doing that?

Reply:
TODO: create a `pub(crate)` function to create instances of the `PositionMonitor`, and make the fields here non-public. This would enforce correct usage of the `PositionMonitor` after it's being created.